cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -351,22 +358,31 @@ Future<Void> clusterOperatorSecret(Supplier<Date> dateSupplier) {
      * due to a new CA key. It is not necessary when the CA certificate is replace while retaining the existing key.
      */
     Future<Void> rollingUpdateForNewCaKey() {
+        //Maintaing both list of reasons and RestartReasons object, as the first is used for ZK, the second for Kafka
+        //This will be unified in a subsequent PR
         List<String> reason = new ArrayList<>(2);
+        RestartReasons kafkaPodRollReasons = RestartReasons.empty();
 
         if (clusterCa.keyReplaced()) {
             reason.add("trust new cluster CA certificate signed by new key");
+            kafkaPodRollReasons.add(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED);
         }
 
         if (clientsCa.keyReplaced()) {
             reason.add("trust new clients CA certificate signed by new key");
```

Comment:
I don't think ZooKeeper needs to be restarted when clients CA changes. So is this really needed?

Reply:
I'll double check that :)